File: terceiroPeriodo/aeds-ii/tps/tp-i/ex02/Palindromo.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int ehPalindromo(char *word){
    int resp = 0;

    for(int i = 0, j = strlen(word) - 1; i < strlen(word); i++, j--){

        if(word[i] == word[j]){
            resp = 1;
        }else{
            i = strlen(word);
            resp = 0;
        }

    }

    return resp;
}

int main(){
    char word[255];

    fgets(word,255,stdin);
    word[strlen(word) - 1] = '\0';

    if(ehPalindromo(word)){
        printf("SIM\n");
    }else{
        printf("NÃO\n");
    }

    return 0;
}

File: terceiroPeriodo/aeds-ii/tps/tp-i/ex01/Palindromo.java
package ex01;

import java.util.Scanner;

public class Palindromo{

    static Boolean ehPalindromo(String word){

        Boolean resp = false;

        for(int i = 0, j = word.length() - 1; i < word.length(); i++, j--){
            if(word.charAt(i) == word.charAt(j)){
                resp = true;
            }else{
                i = word.length();
                resp = false;
            }
        }

        return resp;
    }
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);

        String word = sc.nextLine();

        if(ehPalindromo(word)){
            System.out.println("SIM");
        }else{
            System.out.println("NÃO");
        }

        sc.close();        
    }
}